proto/engine/v1/json_marshal_unmarshal.go

```diff
@@ -7,6 +7,27 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
+// PayloadIDBytes defines a custom type for Payload IDs used by the engine API
+// client with proper JSON Marshal and Unmarshal methods to hex.
+type PayloadIDBytes [8]byte
```

Comment:
Custom marshaler to ensure that payload ID translates nicely into a hex string when sent via JSON-RPC to geth